com.unity.render-pipelines.core/Editor/SpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,181 @@
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+namespace UnityEditor.Rendering
+{
+	/// <summary>
+	/// Material upgrader and relevant utilities for SpeedTree8.
+	/// </summary>
+	public class SpeedTree8MaterialUpgrader : MaterialUpgrader
+    {
+        private enum WindQuality
+        {
+            None = 0,
+            Fastest,
+            Fast,
+            Better,
+            Best,
+            Palm,
+            Count
+        }
+
+        private static string[] WindQualityString =
+        {
+            "_WINDQUALITY_NONE",
+            "_WINDQUALITY_FASTEST",
+            "_WINDQUALITY_FAST",
+            "_WINDQUALITY_BETTER",
+            "_WINDQUALITY_BEST",
+            "_WINDQUALITY_PALM"
+        };
+
+        /// <summary>
+        /// Creates a material upgrader with only the renames in common between HD and Universal.
+        /// </summary>
+        /// <param name="sourceShaderName">Original ST8 shader name.</param>
+        /// <param name="destShaderName">New ST8 shader name.</param>
+        /// <param name="finalizer">A delegate that postprocesses the material as needed by the render pipeline.</param>
```

Comment:
"as needed" is a bit imprecise. 
Would this rewrite be correct?
`A delegate that postprocesses the material for the render pipeline in use.`